### PR TITLE
feature: finish the CLI logs part

### PR DESCRIPTION
**Pull request by fuweid**
Signed-off-by: Wei Fu <fhfuwei@163.com>

### Ⅰ. Describe what this PR did

Finish the CLI logs part in the pouch 

### Ⅱ. Does this pull request fix one issue?

Fixed https://github.com/alibaba/pouch/issues/1371

### Ⅲ. Describe how you did it

Since the API parts has been done, the CLI part is just to fill the logic. However, during the testing, we found the IO issue like #1471 .

Some case maybe fail and it will be flaky case if #1471 doesn't merge.

### Ⅳ. Describe how to verify it

```
➜  pouch run -d busybox sh -c 'for i in $(seq 1 3); do sleep 1; echo hello-$i; done;'
7ba6e5b25b961841145fd8aacaa4d9269bb38ef55519cf74265480326fd779d9
➜  pouch logs --timestamps 7ba6e5b25b961841145fd8aacaa4d9269bb38ef55519cf74265480326fd779d9
2018-06-05T14:05:31.082112571Z hello-1
2018-06-05T14:05:32.084694723Z hello-2
2018-06-05T14:05:33.09511301Z hello-3
➜  pouch logs --since 2018-06-05T14:05:32Z --until 2018-06-05T14:05:32.084694724Z 7ba6e5b25b961841145fd8aacaa4d9269bb38ef55519cf74265480326fd779d9
hello-2
➜  pouch logs --until 2018-06-05T14:05:32.084694724Z 7ba6e5b25b961841145fd8aacaa4d9269bb38ef55519cf74265480326fd779d9
hello-1
hello-2
```

### CODE REVIEW

**Comment by allencloud** [comment]
Unfortunately, CI fails with golint, @fuweid :
```
/go/bin/golint
pkg/utils/utils.go:141:4: should replace numColons -= 1 with numColons--
Makefile:57: recipe for target 'lint' failed
```
Interesting.

**Comment by fuweid** [comment]
@allencloud I want to add the `make check` in the git pre-hook. just reminder haha

**Comment by fuweid** [comment]
@HusterWan Please help me to check the failed case 😄 thanks

**Comment by allencloud** [comment]
Quite interesting, the test coverage decreases. 😄

File: daemon/logger/jsonfile/utils.go [hunk 1]
```diff
@@ -193,15 +193,19 @@ func seekOffsetByTailLines(rs io.ReadSeeker, n int) (int64, error) {
...
 	for {
+		readN = int64(blockSize)
 		left = size + int64(block*blockSize)
 		if left < 0 {
+			readN = int64(blockSize) + left
 			left = 0
 		}
 
-		b = make([]byte, blockSize)
+		b = make([]byte, readN)
```
Comment (zhuangqh) [inline daemon/logger/jsonfile/utils.go]: If the content don't end with `\n` like `1\n2\n3\n4`, `seekOffsetByTailLines(, 2)` will return the index of `2`. The tail content would be `2\n3\n4`. Compared with command tail, `echo "1\n2\n3\n4" > a.txt; tail -n2 a.txt` is `3\n4`.
Does this function works as expected?😁 @fuweid

**Comment by zhuangqh** [comment]
haha. I got it. Thanks for your reply at midnight~ @fuweid

**Comment by allencloud** [comment]
LGTM

**Pull request merged**